lib/Ama/Controller/Feedback.pm

```diff
@@ -4,15 +4,20 @@ use Mojo::Base 'Mojolicious::Controller';
 sub submit{
   my $self = shift;
   my $sendgrid = $self->sendgrid;
+  my $send;
   my $feedback = $self->param('feedback');
-  my $send = $sendgrid->mail(
-               text    => $feedback,
-               to      => $self->config->{sendgrid}->{to},
-               cc      => $self->config->{sendgrid}->{cc},
-               bcc     => $self->config->{sendgrid}->{bcc},
-               from    => $self->config->{sendgrid}->{from},
-               subject => "feedback received from AMA",
-             )->send;
+  if (!$self->config->{sendgrid}->{to} || $self->config->{sendgrid}->{to} eq undef || !$self->config->{sendgrid}->{from} || $self->config->{sendgrid}->{from} eq undef){
+                warn "missing to or from address!\n";
```

Comment:
Good start, but at this point the application is already running therefore warnings and errors need to get reported to the user.

You should also do a check at application start up (lib/Ama.pm) that checks for certain requirements, and if missing, either A) bail, or B) warn and disable the functionality.